room.cpp

```diff
@@ -24,6 +24,8 @@
 #include <QtCore/QJsonArray>
 #include <QtCore/QStringBuilder> // for efficient string concats (operator%)
 #include <QtCore/QElapsedTimer>
+#include <algorithm>
+#include <array>
```

Comment:
You don't need these two. `<algorithm>` is always included through Qt headers (take a look at `qglobal.h`) and `<array>` is already included a few lines above